packages/core/src/ActionsGeneric/ActionsGeneric.tsx

```diff
@@ -2,41 +2,55 @@ import React, { isValidElement } from "react";
 import { MoreOptionsVertical } from "@hitachivantara/uikit-react-icons";
 
 import { useDefaultProps } from "../hooks/useDefaultProps";
-import { HvButton, HvButtonVariant } from "../Button";
+import { HvButton, HvButtonProps, HvButtonVariant } from "../Button";
 import { HvDropDownMenu } from "../DropDownMenu";
 import { setId } from "../utils/setId";
 import { ExtractNames } from "../utils/classes";
 import { HvBaseProps } from "../types/generic";
-
 import { staticClasses, useClasses } from "./ActionsGeneric.styles";
+import { IconButton } from "../utils/IconButton";
 
 export { staticClasses as actionsGenericClasses };
 
 export type HvActionsGenericClasses = ExtractNames<typeof useClasses>;
 
 export interface HvActionGeneric {
+  /** Action id. */
   id: string;
+  /** Action label. */
   label: string;
+  /** Action icon. */
   icon?:
     | React.ReactNode
     | ((params: { isDisabled?: boolean }) => React.ReactNode);
+  /** Whether the action is disabled or not. */
   disabled?: boolean;
+  /** When set to `true`, the button will have the icon has its content and a tooltip with the label will appear when the button is visible and hovered. */
+  iconButton?: boolean;
 }
 
 export interface HvActionsGenericProps extends HvBaseProps {
-  /** Button category. */
+  /**
+   * The button category for all actions.
+   *
+   * @deprecated Use `variant` instead.
+   */
   category?: HvButtonVariant;
-  /**  Whether actions should be all disabled */
+  /** The button variant for all actions. */
+  variant?: HvButtonVariant;
+  /** Whether the actions should be all disabled. */
   disabled?: boolean;
-  /** The renderable content inside the actions slot of the footer, or an Array of actions `{id, label, icon, disabled}` */
+  /** Whether the actions should be all icon buttons when visible. */
+  actionsIconButton?: boolean;
+  /** The renderable content inside the actions slot of the footer, or an array of actions. */
   actions: React.ReactNode | HvActionGeneric[];
-  /**  The callback function ran when an action is triggered, receiving `action` as param */
+  /** The callback function called when an action is triggered, receiving the `action` as parameter. */
   actionsCallback?: (
```

Comment:
Yeah, it makes sense. I'll make this change in this PR.

Reply:
Several other components are also using `actionsCallback` (since they use `HvActionsGeneric` internally). To not go beyond the scope of this PR, I'll deprecate the prop in the other components in another PR. For now, I deprecated it in `HvActionsGeneric` and the Flow component.

Reply:
Definitely! I meant only for `HvFlow`, as these are new props - we can refactor the rest later